include/sparrow/dictionary_encoded_layout.hpp

```diff
@@ -22,41 +22,66 @@
 
 namespace sparrow
 {
-    /**
-     * @class dictionary_value_iterator
-     *
-     * @brief Iterator over the data values of a dictionary layout.
-     *
-     * @tparam IL the layout type of the indexes.
-     * @tparam SL the layout type of the dictionary.
-     * @tparam is_const a boolean flag specifying whether this iterator is const.
-     */
-    template <class IL, class SL, bool is_const>
-    class dictionary_value_iterator : public iterator_base<
-                                          dictionary_value_iterator<IL, SL, is_const>,
-                                          typename SL::value_type,
-                                          std::random_access_iterator_tag,
-                                          typename SL::const_reference>
+    template <class L, bool IC>
+    struct dictionary_value_traits
+    {
+        using layout_type = L;
+        using value_type = typename L::inner_value_type;
+        using tag = std::random_access_iterator_tag;
+        using const_reference = typename L::inner_const_reference;
+        static constexpr bool is_value = true;
+        static constexpr bool is_const = IC;
+    };
+
+    template <class L, bool IC>
+    struct dictionary_bitmap_traits
+    {
+        using layout_type = L;
+        using value_type = typename L::bitmap_value_type;
+        using tag = std::random_access_iterator_tag;
+        using const_reference = typename L::bitmap_const_reference;
+        static constexpr bool is_value = false;
+        static constexpr bool is_const = IC;
+    };
+
+    template <class Traits>
```

Comment:
Constraining the traits parameter would be helpful (something like `requires is_instance_of_v<dictionary_bitmap_traits>` would be enough I guess?)

Reply:
It's either bitmap or value traits